### DISCUSSION TITLE
Convex decomposition

### DISCUSSION THREAD

**Discussion by indigoyeoma** [haosulab/SAPIEN#127] [Q&A]
Hello, I am learning to use your SAPIEN simulator for manipulation, and have been amazed at your features.

Though Im struggling with a problem, and wonder if I could get some help.

I am trying to load a cup in the simulator and try to grasp the cup by the handle or the rim
But I would want a more detailed convex hull to gain the rim of the cup  and the hole of the handle for the task
I have read the paper and mentioned that there is a way for convex decomposition, but could not find the functions in the docs.

Can you help me on how to use the  convex decomposition?

**Comment by fbxiang** [comment]
For convex decomposition, I recommend checking out our convex decomposition library https://github.com/SarahWeiii/CoACD. This library will be directly integrated into the next major release of SAPIEN (SAPIEN 3). The feature is currently implemented in the dev branch.
https://github.com/haosulab/SAPIEN/blob/8fd651728c40047337e162c0199322b86dc76655/python/py_package/wrapper/actor_builder.py#L447-L448
